libguard/guard_interface.cpp

```diff
@@ -169,6 +188,9 @@ GuardRecord create(const EntityPath& entityPath, uint32_t eId, uint8_t eType)
         lastPos++;
     }
 
+	// Space left in file before writing new record
+    fileSizeCheck(file.size(), lastPos + 1);
```

Comment:
can you delete resolved record (invalid id) when you don't have space?
OR the customer need to trigger deleteAll? If doing the deleteAll then it will delete all or only resolved record?

Need some documentation about what action customer needs to do or libguard won't do such an activity and the consumer of the libguard needs to decide when the guard file is overflowing?

Reply:
We can delete the invalid guard records. If we are handling this at libguard end than no need of throwing exception saying there is no space left in file. We need to decide on this.

Reply:
- Where you have documented that information? what is the action you won't do when a file is fulled so that consumers can decide what needs to be done? - I think we cannot say what need by done by the caller, its up to them/

> If we are handling this at libguard end than no need of throwing
Why so, it will drive based on what you are going to the delete operation let say, the file is fulled only with unresolved records - what you will do ?

Reply:
As per discussion, will delete the invalid/resolved guard records when space is full.